lib/improver/ensemble_copula_coupling/ensemble_copula_coupling_utilities.py

```diff
@@ -147,3 +176,110 @@ def create_cube_with_percentiles(percentiles, template_cube, cube_data):
         dims = tuple([dim+1 for dim in dims])
         result.add_aux_coord(coord.copy(), dims)
     return result
+
+
+def get_bounds_of_distribution(cube_name, cube_units):
+    """
+    Gets the bounds of the distribution and converts the units of the
+    bounds_pairing to the units of the forecast.
+
+    This method gets the bounds values and units from the imported
+    dictionaries: bounds_for_ecdf and units_of_bounds_for_ecdf.
+    The units of the bounds are converted to be the units of the input
+    cube.
+
+    Parameters
+    ----------
+    cube_name : String
+        Name of cube, which is used as the key for the bounds_for_ecdf and
+        units_of_bounds_for_ecdf dictionaries.
+    cube_units : cf_units.Unit
+        Units to which the bounds_pairing will be converted.
+
+    Returns
+    -------
+    bounds_pairing : Tuple
+        Lower and upper bound to be used as the ends of the
+        empirical cumulative distribution function, converted to have
+        the same units as the input cube.
+
+    """
+    # Extract bounds from dictionary of constants.
+    try:
+        bounds_pairing = bounds_for_ecdf[cube_name]
+        bounds_pairing_units = (
+            units_of_bounds_for_ecdf[cube_name])
+    except KeyError as err:
+        msg = ("The forecast_cube name: {} is not recognised"
+               "within bounds_for_ecdf {} or "
+               "units_of_bounds_for_ecdf: {}. \n"
+               "Error: {}".format(
+                   cube_name, bounds_for_ecdf,
+                   units_of_bounds_for_ecdf, err))
+        raise KeyError(msg)
+    bounds_pairing_units = unit.Unit(bounds_pairing_units)
+    bounds_pairing = bounds_pairing_units.convert(
+        np.array(bounds_pairing), cube_units)
+    return bounds_pairing
+
+
+def insert_lower_and_upper_endpoint_to_1d_array(
+        array_1d, low_endpoint, high_endpoint):
+    """
+    For a 1d array, add a lower and upper endpoint.
+
+    Parameters
+    ----------
+    array_1d : Numpy array
+        1d array of values
+    low_endpoint : Number
+        Number of use as the lower endpoint.
+    high_endpoint : Number
+        Number of use as the upper endpoint.
+    """
+    array_1d = np.insert(array_1d, 0, low_endpoint)
+    array_1d = np.append(array_1d, high_endpoint)
+    return array_1d
+
+
+def reshape_array_to_have_probabilistic_dimension_at_the_front(
```

Comment:
Factored out as this functionality was very similar in a few places in lib/improver/ensemble_copula_coupling/ensemble_copula_coupling.py